tools/cellxgene_census_builder/src/cellxgene_census_builder/__main__.py

```diff
@@ -38,31 +38,63 @@ def main() -> int:
     # Process initialization/setup must be done early. NOTE: do NOT log before this line!
     process_init(build_args)
 
-    # Return process exit code (or raise, which exits with a code of `1`)
-    return do_build(build_args, skip_completed_steps=cli_args.test_resume)
+    # List of available commands for the builder
+    commands = {
+        "release": [do_the_release],
+        "replicate": [do_sync_artifact_to_replica_s3_bucket],
+        "sync-release": [
+            do_sync_release_file_to_replica_s3_bucket,
+        ],
+        "build": [
+            do_prebuild_set_defaults,
+            do_prebuild_checks,
+            do_build_soma,
+            do_validate_soma,
+            do_create_reports,
+            do_data_copy,
+            do_report_copy,
+            do_log_copy,
+        ],
+        "cleanup": [
+            do_old_release_cleanup,
+        ],
+        "full-build": [
+            do_prebuild_set_defaults,
+            do_prebuild_checks,
+            do_build_soma,
+            do_validate_soma,
+            do_create_reports,
+            do_data_copy,
+            do_the_release,
+            do_report_copy,
+            do_old_release_cleanup,
+            do_log_copy,
+        ],
+    }
 
+    # The build command is set via the CENSUS_BUILD_COMMAND environment variable.
+    # If the variable is not defined, perform a full build by default.
+    command = os.getenv("CENSUS_BUILD_COMMAND", "full-build")
 
-def do_build(args: CensusBuildArgs, skip_completed_steps: bool = False) -> int:
-    """
-    Top-level build sequence.
+    # Used for testing.
+    if command == "pass":
+        return 0
+
+    if command not in commands:
+        logging.critical(f"Unknown command: {command}")
+        return 1
 
-    Built steps will be executed in order. Build will stop if a build step returns non-zero
-    exit code or raises.
+    build_steps = commands[command]
+    rv = _do_steps(build_steps, build_args, cli_args.test_resume)
+    return rv
+
+
+def _do_steps(
+    build_steps: Sequence[Callable[[CensusBuildArgs], bool]], args: CensusBuildArgs, skip_completed_steps: bool = False
```

Comment:
By running this in an ephemeral environment, this isn't really used anymore.

Reply:
ok, yeah, I haven't thought through what ad hoc testing looks like now--maybe we could a few sentences about how to manually test the multi-step process in dev env on an instance w/o StepFn orchestration.

Reply:
It is still an issue when there is a failure of some sort, not caught correctly by the code.  I am a big fan of belt-and-suspenders at this level.

In other words, perhaps the right behavior is to _error_ out, rather than skipping.

Reply:
Will add the checks.